sound/soc/sof/intel/hda-dsp.c

```diff
@@ -375,6 +396,18 @@ static int hda_resume(struct snd_sof_dev *sdev)
 	snd_hdac_ext_bus_ppcap_enable(bus, true);
 	snd_hdac_ext_bus_ppcap_int_enable(bus, true);
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC)
+	if (runtime_resume) {
+		/* check jack status based on controller status */
+		hda_codec_jack_poll(sdev, status);
+
+		/* disable controller Wake Up event*/
+		snd_hdac_chip_writew(bus, WAKEEN,
+				     snd_hdac_chip_readw(bus, WAKEEN) &
+				     ~STATESTS_INT_MASK);
+	}
+#endif
```

Comment:
you can use a helper that does nothing when HDA_AUDIO_CODEC is not defined. This would make the code more readable